scene/gui/grid_container.cpp

```diff
@@ -242,14 +242,17 @@ Size2 GridContainer::get_minimum_size() const {
 }
 
 Control *GridContainer::get_child_control_at_cell(int row, int column) {
-	Control *c;
+	Control *c = NULL;
 	int grid_index = row * columns + column;
 	for (int i = 0; i < get_child_count(); i++) {
 		c = Object::cast_to<Control>(get_child(i));
-		if (!c || !c->is_visible_in_tree())
+		if (!c || !c->is_visible_in_tree()) {
+			c = NULL;
+			grid_index++;
```

Comment:
See my comment on https://github.com/godotengine/godot/pull/16977#issuecomment-426972105, I think this method is bogus, and unless I'm mistaken, this fix would make it even more wrong.

Sounds like this methof should just be:
```
int grid_index = row * columns + column;
Control *c = Object::cast_to<Control>(get_child(grid_index));
return c;
```

But I'd be more for reverting that PR honestly.

Reply:
I'm going to revert it in #22822 so you can rebase this commit away.

Reply:
Okay, I think the logic behind that function was getting the element at row X, column Y.
Hidden elements to not take grid space, so they must be skipped.
Non-control elements do not take grid space either, so they must be skipped.

So:
```
int grid_index = row * columns + column;
Control *c = Object::cast_to<Control>(get_child(grid_index));
```
Will not give you the element you desire if there is any hidden element, or non control element as a child.
The original logic was broken, yes, but I think the change I did was actually fixing it.
The idea is:
```
If element is not found, return NULL
Get the grid_index
Loop through children
  If element is not control or is hidden, skip it and raise grid_index (if element was 2, but it's hidden, we want 3)
  else if the current child is the one we are looking for (`i == grid_index`) break the loop
Return the found element, or null if none was found (i.e. there is no element at `row` `column`).
[should probably FAIL_COND_V(column >= comlums, NULL)]
```
Anyway, this probably deserve a new PR if we deem this method necessary, so I will remove it from this PR.

Reply:
Ah right, I understand now. It's not a trivial operation then and I understand the need for a for loop. Might be worth readding with your fixed implementation and more comments in the code to make clear that non-Control and non-visible elements do not take up a cell and should thus be skipped.